op-chain-ops/crossdomain/withdrawal.go

```diff
@@ -130,3 +141,100 @@ func (w *Withdrawal) StorageSlot() (common.Hash, error) {
 	slot := crypto.Keccak256(preimage)
 	return common.BytesToHash(slot), nil
 }
+
+// Compute the receipt corresponding to the withdrawal. This receipt
+// is in the bedrock transition block. It contains 3 logs.
+// SentMessage, SentMessageExtension1 and MessagePassed.
+// These logs are enough for the standard withdrawal flow to happen
+// which is driven by events being emitted.
+func (w *Withdrawal) Receipt(hdr *types.Header) (*types.Receipt, error) {
+	// Create a new receipt with the state root, successful execution and no gas
+	// used
+	receipt := types.NewReceipt(hdr.Root.Bytes(), false, 0)
+	if receipt.Logs == nil {
+		receipt.Logs = make([]*types.Log, 0)
+	}
+
+	// Create the SentMessage log.
+	args := abi.Arguments{
+		{Name: "target", Type: AddressType},
+		{Name: "sender", Type: AddressType},
+		{Name: "data", Type: BytesType},
+		{Name: "nonce", Type: Uint256Type},
+	}
+	data, err := args.Pack(w.Target, w.Sender, w.Data, w.Nonce)
+	if err != nil {
+		return nil, err
+	}
+	// The L2CrossDomainMessenger emits this event. The target is
+	// indexed.
+	sm := &types.Log{
+		Address: predeploys.L2CrossDomainMessengerAddr,
+		Topics: []common.Hash{
+			SentMessageEventABIHash,
+			w.Target.Hash(),
+		},
+		Data:        data,
+		BlockNumber: hdr.Number.Uint64(),
+		TxHash:      common.Hash{},
+		TxIndex:     0,
+		BlockHash:   hdr.Hash(),
+		Index:       0,
```

Comment:
I'm a little worried about using the log + transaction index.

Reply:
I set the tx index to 0 as well as the index to 0 for all logs. It is unexpected behavior to have the index be the same for each log, but I don't think it will break anything. I could refactor things such that the index is non zero and increments

Reply:
i've canonicalized the tx index + log index

Reply:
thanks for doing this